src/main/java/entity/Note.java

```diff
@@ -1,4 +1,70 @@
 package entity;
 
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+/**
+ * Class representing a Note entity
+ * This class is used for adding an extra note to references. 
+ * @author Andrej Staruch
+ */
+
+@Entity
+@Table(name = "Notes")
 public class Note {
+    @Id
+    @GeneratedValue
+    private Long id;
+    
+    private String text;
+
+    private User creator;
+    
+    @OneToMany(mappedBy = "references")
+    private Reference reference;
```

Comment:
Ta vazba tam taky nemá být asi, protože ji budeš mít na Referenci, to by mělo stačit. Tzn. k těm poznámkám budeš přistupovat přes referenci, naopak to stejně nedává smysl.